### PR TITLE
[WIP] Update to Sphinx-Gallery 0.1.3

### PR DESCRIPTION
**Pull request by Titan-C**
I recently released a new version of Sphinx-Gallery. It has quite a bunch of bug fixes and the new feature of exiting with fail if examples did not run successfully.

### CODE REVIEW

**Comment by KamalakerDadi** [comment]
I often see this CircleCI failure in my PR's as well in other PR's. Still I don't know the reason for that failure.

**Comment by Titan-C** [comment]
The second commit was intended to fail in CircleCI because I included a failing example and that it shall stop as soon as it is encountered. Does happen. Now There should be the traceback of that fail, but somehow is not printed at the end but a quite a bit earlier. Is the RuntimeError.

Then the error why the image is missing is still a mystery to me.

**Comment by Titan-C** [comment]
I know why the error is not at the end. It is a problem on the redirection of stderr into stdout.

**Comment by GaelVaroquaux** [comment]
> Then the error why the image is missing is still a mystery to me.

Yes, I plan to try to debug this today. It's been failing lately in
master. I can't reproduce it localy :/

**Comment by lesteve** [comment]
>>  Then the error why the image is missing is still a mystery to me.

> Yes, I plan to try to debug this today. It's been failing lately in
master. I can't reproduce it localy :/

Missing images generally are a symptom of an error in an example. Looking at the first failure in master (https://circleci.com/gh/nilearn/nilearn/1880), the root problem seems to be:

```
../examples/03_connectivity/plot_connectivity_measures.py is not compiling:
Traceback (most recent call last):
  File "/home/ubuntu/nilearn/doc/sphinxext/sphinx_gallery/gen_rst.py", line 467, in execute_script
    exec(code_block, example_globals)
  File "<string>", line 8, in <module>
  File "/home/ubuntu/miniconda2/envs/testenv/lib/python3.5/site-packages/sklearn/base.py", line 455, in fit_transform
    return self.fit(X, **fit_params).transform(X)
  File "/home/ubuntu/nilearn/nilearn/connectome/connectivity_matrices.py", line 348, in fit
    self.mean_ = _geometric_mean(covariances, max_iter=30, tol=1e-7)
  File "/home/ubuntu/nilearn/nilearn/connectome/connectivity_matrices.py", line 141, in _geometric_mean
    _check_spd(matrix)
  File "/home/ubuntu/nilearn/nilearn/connectome/connectivity_matrices.py", line 36, in _check_spd
    raise ValueError('Expected a symmetric positive definite matrix.')
ValueError: Expected a symmetric positive definite matrix.
```

**Comment by AlexandreAbraham** [comment]
I have no failed example on my box :-/

**Comment by lesteve** [comment]
> I have no failed example on my box :-/

Same for me, I tried doing a make clean and make html-strict earlier today and it worked fine ...

**Comment by lesteve** [comment]
Actually I think @GaelVaroquaux has spotted it and this is related to scipy 0.18. Upgrading scipy to 0.18 I get the same error as in CircleCI.

**Comment by GaelVaroquaux** [comment]
> Same for me, I tried doing a make clean and make html-strict earlier
> today and it worked fine ...

It's a change in scipy 0.18 (I am a bit unhappy about that):
https://github.com/nilearn/nilearn/pull/1199

Working on fixing that. Give me another hour.

**Comment by GaelVaroquaux** [comment]
@Titan-C : do you mind rebasing this PR on master, to benefit from the fix that I made for circleCI to run.

**Comment by Titan-C** [comment]
I rebased to master with only a clean update of the new Sphinx-Gallery. The test on the early abort with failing exitcode worked.
Because of the html-strict you use to build nilearn I could not test the case of a failing example that still finished the complete build but exited with error to CircleCI. I'll commit a test to that later if this build does not fail. If that test prove to give a good reliability to inform that examples failed I'll remove the current grep test on the current `.circle.yml` file

**Comment by lesteve** [comment]
> Because of the html-strict you use to build nilearn I could not test the case of a failing example that still finished the complete build but exited with error to CircleCI. I'll commit a test to that later if this build does not fail.

Sorry I don't understand what you mean.

**Comment by lesteve** [comment]
> If that test prove to give a good reliability to inform that examples failed I'll remove the current grep test on the current .circle.yml file

This would be good indeed to remove the `grep Traceback` as a way to make sure that no examples has failed.

**Comment by GaelVaroquaux** [comment]
@Titan-C : I'd love to merge this, but the continuous integration (specifically circle-ci) is failing, and it seems like a genuine problem with sphinx-gallery. Could you have a look please? Thanks!

**Comment by Titan-C** [comment]
So the test to fail the build on purpose was successful. Sphinx-Gallery does exit with fail after an example has not run. I'm deleting such test from the commit log now.

**Comment by Titan-C** [comment]
>This would be good indeed to remove the grep Traceback as a way to make sure that no examples has failed.

Now that Sphinx-Gallery does exit with fail on broken examples already in https://github.com/nilearn/nilearn/blame/master/circle.yml#L35
I see no reason to keep this:
https://github.com/nilearn/nilearn/blame/master/circle.yml#L38-L44
Any comments on that @chrisfilo , @aabadie , @lesteve

**Comment by GaelVaroquaux** [comment]
@Titan-C: this is good to go, as far as I am concerned.

One last question: how easy is it for you to embed the dev version of
Sphinx-Gallery, in order to have the latest CSS?

**Comment by Titan-C** [comment]
> One last question: how easy is it for you to embed the dev version of Sphinx-Gallery, in order to have the latest CSS?

Not too hard. Just pushed it

File: doc/sphinxext/sphinx_gallery/__init__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 """
 import os
-__version__ = '0.1.3'
+__version__ = '0.1.4-dev'
```
Comment (GaelVaroquaux) [inline doc/sphinxext/sphinx_gallery/__init__.py]: Strange, the changes to gen_rst.py are missing from this commit, which implies that the download buttons don't render right. I'll merge and add them manually. Thanks

**Comment by GaelVaroquaux** [comment]
Looks good. I am merging. Thanks heaps.

It seems to me that gen_rst.py got out of sync with sphinx-gallery master. I'll fix that.

**Pull request merged**